convert2rhel/checks.py

```diff
@@ -75,77 +89,79 @@ def perform_pre_ponr_checks():
-        "name=Convert2RHEL Repository\n"
-        "baseurl=https://cdn.redhat.com/content/public/convert2rhel/$releasever/x86_64/os/\n"
-        "gpgcheck=1\n"
-        "enabled=1\n"
-        "sslcacert=%s\n"
-        "gpgkey=file://%s\n" % (ssl_cert_path, gpg_path)
-    )
+    """Make sure that we are running the latest downstream version of convert2rhel"""
+    logger.task("Prepare: Checking if this is the latest version of convert2rhel")
+
+    if not system_info.has_internet_accsess:
+        logger.warning("Skipping the check because no internet connection has been detected.")
+        return
 
     cmd = [
         "repoquery",
         "--releasever=%s" % system_info.version.major,
     ]
 
-    repo_dir = tempfile.mkdtemp(prefix="convert2rhel_repo.", dir=utils.TMP_DIR)
-    utils.mkdir_p(repo_dir)
-    repo_path = os.path.join(repo_dir, "convert2rhel.repo")
-    store_content_to_file(filename=repo_path, content=repo_content)
-
-    cmd.append("--setopt=reposdir=%s" % repo_dir)
+    cmd.append(
+        "--repofrom='convert2rhel,https://cdn.redhat.com/content/public/convert2rhel/$releasever/$basearch/os/'convert2rhel"
+    )
     cmd.append("convert2rhel")
 
     raw_output_convert2rhel_versions, return_code = run_subprocess(cmd, print_output=False)
 
     if return_code != 0:
         logger.warning(
-            "Couldn't check if this is the latest version of convert2rhel\n"
-            "repoquery failed %s, %s" % (return_code, raw_output_convert2rhel_versions)
+            "Couldn't check if currently installed convert2rhel is latest version.\n"
+            "repoquery failed with the following output:\n%s" % (raw_output_convert2rhel_versions)
         )
         return
 
     PKG_NEVR = r"\b(\S+)-(?:([0-9]+):)?(\S+)-(\S+)\b"
     convert2rhel_versions = re.findall(PKG_NEVR, raw_output_convert2rhel_versions, re.MULTILINE)
-    latest_version = ("0", "0.00", "0")
+    logger.debug("Found %s convert2rhel package(s)" % len(convert2rhel_versions))
+    latest_available_version = ("0", "0.00", "0")
 
+    # This loop will determine the latest available convert2rhel version in the yum repo.
+    # It assigns the epoch, version, and release ex: ("0", "0.26", "1.el7") to the latest_available_version variable.
     for package_version in convert2rhel_versions:
-        # rpm.lableCompare(pkg1, pkg2) compare two kernel version strings and return
-        # -1 if str2 is greater then str1, 0 if they are equal, 1 if str1 is greater the str2
-        ver_compare = rpm.labelCompare(package_version[1:], latest_version)
+        # rpm.labelCompare(pkg1, pkg2) compare two package version strings and return
+        # -1 if latest_version is greater than package_version, 0 if they are equal, 1 if package_version is greater than latest_version
+        ver_compare = rpm.labelCompare(package_version[1:], latest_available_version)
         if ver_compare > 0:
-            latest_version = package_version[1:]
+            logger.debuge(
```

Comment:
```suggestion
            logger.debug(
```

Reply:
Thank you for spotting this, I will go ahead and get this changed then push it with the unit test!